resource-managers/kubernetes/core/src/test/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorLifecycleTestUtils.scala

```diff
@@ -106,6 +106,31 @@ object ExecutorLifecycleTestUtils {
       .build()
   }
 
+  /**
+   * This creates a pod with a finished executor and running sidecar
+   */
+  def finishedExecutorWithRunningSidecar(
```

Comment:
This looks like a test helper because this is used in the test case only. In that case, shall we move this to `ExecutorPodsSnapshotSuite.scala`?

Reply:
I don't have a strong preference, this is the other place where the utils functions for `ExecutorPodsSnapshotSuite` live though so I was going for consistency.

Reply:
Got it.